File: 1/b.ts
const input = await Bun.file("input.txt").text();
// const input = await Bun.file("test.txt").text();

const digits = [
  "one",
  "two",
  "three",
  "four",
  "five",
  "six",
  "seven",
  "eight",
  "nine",
];

const replaceDigitNames = (s: string) => {
  for (let i = 0; i < digits.length; i++) {
    s = s.replaceAll(digits[i], `${digits[i]}${i + 1}${digits[i]}`);
  }
  return s;
};

const lines = input.split("\n").map((line) => {
  const d = replaceDigitNames(line);
  console.log(d);
  return Array.from(d)
    .map((d) => parseInt(d))
    .filter((d) => !isNaN(d));
});

let acc = 0;

for (const line of lines) {
  acc += line[0] * 10 + line.at(-1)!;
}

console.log(acc);

File: 1/b.go
package main

import (
	"fmt"
	"os"
	"strconv"
	"strings"
)

func main() {
	numberNames := []string{"zero", "one", "two", "three", "four", "five", "six", "seven", "eight", "nine"}

	file, err := os.ReadFile("input.txt")
	if err != nil {
		panic(err)
	}
	lines := strings.Split(string(file), "\n")
	acc := 0
	for _, line := range lines {
		first := -1
		last := -1
		for i, name := range numberNames {
			line = strings.ReplaceAll(line, name, name+strconv.Itoa(i)+name)
		}
		for _, char := range line {
			number, err := strconv.Atoi(string(char))
			if err == nil {
				if first == -1 {
					first = number
				}
				last = number
			}
		}
		if first == -1 {
			continue
		}
		fmt.Println(first, last)
		acc += first*10 + last
	}
	fmt.Println(acc)
}
